File: server/routers/market.py
"""
市场数据路由
"""
from fastapi import APIRouter
from typing import Optional

from services.eastmoney_service import (
    get_eastmoney_service,
    IndexQuote,
    StockQuote,
    MarketSentiment,
)

router = APIRouter()


@router.get("/overview")
async def get_market_overview():
    """获取市场概览"""
    service = get_eastmoney_service()
    return await service.get_market_overview()


@router.get("/indices", response_model=list[IndexQuote])
async def get_indices():
    """获取主要指数行情（上证、深证、创业板、科创、沪深300、上证50）"""
    service = get_eastmoney_service()
    return await service.get_indices()


@router.get("/sentiment", response_model=MarketSentiment)
async def get_sentiment():
    """获取市场情绪数据（涨停、跌停、冲板、炸板率、情绪等）"""
    service = get_eastmoney_service()
    return await service.get_sentiment()


@router.get("/quote/{symbol}", response_model=StockQuote)
async def get_stock_quote(symbol: str):
    """获取单个股票行情"""
    service = get_eastmoney_service()
    quote = await service.get_stock_quote(symbol)
    if quote is None:
        return {"error": "股票不存在或数据获取失败"}
    return quote


@router.get("/limit-up", response_model=list[StockQuote])
async def get_limit_up_stocks():
    """获取涨停股列表"""
    service = get_eastmoney_service()
    return await service.get_limit_up_stocks()


@router.post("/refresh")
async def refresh_data():
    """手动刷新数据"""
    service = get_eastmoney_service()
    await service.refresh_stocks()
    return {"status": "ok", "message": "数据已刷新"}

File: app/src/lib/api/market.ts
/**
 * 市场数据 API
 */
import { api } from './client';

export interface StockQuote {
  symbol: string;
  name: string;
  price: number;
  change: number;
  change_pct: number;
  volume: number;
  amount: number;
  open: number;
  high: number;
  low: number;
  pre_close: number;
  update_time: string;
}

export interface StockInfo {
  symbol: string;
  name: string;
  industry: string;
  market: string;
}

export interface MarketOverview {
  up_count: number;
  down_count: number;
  flat_count: number;
  limit_up_count: number;
  limit_down_count: number;
  total_amount: number;
  north_flow: number;
  update_time: string;
}

export interface IndexQuote {
  code: string;
  name: string;
  price: number;
  change: number;
  change_pct: number;
  update_time: string;
}

export interface MarketSentiment {
  limit_up_count: number;
  limit_down_count: number;
  up_count: number;
  down_count: number;
  flat_count: number;
  rush_count: number;
  bomb_count: number;
  bomb_rate: number;
  max_streak: number;
  sentiment: string;
  total_amount: number;
  update_time: string;
}

export interface KlineData {
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  amount: number;
}

export const marketApi = {
  /** 获取市场概览 */
  getOverview: () => api.get<MarketOverview>('/market/overview'),
  
  /** 获取主要指数行情 */
  getIndices: () => api.get<IndexQuote[]>('/market/indices'),
  
  /** 获取市场情绪数据 */
  getSentiment: () => api.get<MarketSentiment>('/market/sentiment'),
  
  /** 获取股票列表 */
  getStockList: (market?: string) => 
    api.get<StockInfo[]>('/market/stocks', { market }),
  
  /** 获取单个股票行情 */
  getQuote: (symbol: string) => 
    api.get<StockQuote>(`/market/quote/${symbol}`),
  
  /** 批量获取行情 */
  getBatchQuotes: (symbols: string[]) => 
    api.post<StockQuote[]>('/market/quotes', symbols),
  
  /** 获取涨停股列表 */
  getLimitUpStocks: () => 
    api.get<StockQuote[]>('/market/limit-up'),
  
  /** 获取K线数据 */
  getKline: (symbol: string, startDate: string, endDate?: string) =>
    api.get<KlineData[]>(`/market/kline/${symbol}`, { start_date: startDate, end_date: endDate }),
};
